docs/marketplace/add_to_marketplace/declarative_marketplace.md

```diff
@@ -0,0 +1,268 @@
+
+## Setup a Custom Declarative Journey
+
+Here we explain how you can implement a Custom Declarative Journey.
+
+### Requirements
+
+1. Git Provider
+2. CI/CD Tool
+3. a [Service Account](/development_suite/identity-and-access-management/manage-service-accounts.md)
+
+:::info
+
+For a thorough explanation on supported Providers and how to configure them in the Console, [consult this guide](/console/company-configuration/providers/configure-provider.mdx)
+
+:::
+
+### Creation of the repository
+
+Imagine you already have some Marketplace items and their relative file manifest, ready to be deployed to the Console.
+
+Create a repository on your Git Provider, and add the item files following this structure:
+
+```
+Root
+│
+├── 📁 images
+│   ├── 🖼️ product_service.png
+│   ├── 🖼️ user_management.png
+│   ├── 🖼️ payment_gateway.png
+│   ├── 🖼️ recommendation_engine.png
+│   ├── 🖼️ analytics_service.png
+│   ├── 🖼️ nodejs_template.png
+│   ├── 🖼️ python_flask_template.png
+│   ├── 🖼️ java_springboot_template.png
+│   ├── 🖼️ user_management_example.png
+│   ├── 🖼️ ecommerce_application.png
+│   ├── 🖼️ inventory_system.png
+│   ├── 🖼️ user_authentication.png
+│   ├── 🖼️ order_processing.png
+│   ├── 🖼️ notification_service.png
+│   ├── 🖼️ logging_system.png
+|   ├── 🖼️ user_management_example.png
+│   ├── 🖼️ ecommerce_application.png
+│   ├── 🖼️ acme_company.png
+│
+├── 📁 resources
+│   ├── 📁 plugins
+│   │   ├── 📁 inventory_system
+│   │   │   ├── 📄 inventory_system_config.json
+│   │   ├── 📁 user_authentication
+│   │   │   ├── 📄 user_authentication_config.json
+│   │   ├── 📁 order_processing
+│   │   │   ├── 📄 order_processing_config.yml
+│   │   ├── 📁 notification_service
+│   │   │   ├── 📄 notification_service_config.yml
+│   │   ├── 📁 logging_system
+│   │   │   ├── 📄 logging_system_config.json
+├── 📁 templates
+│   ├── 📁 nodejs_template
+│   │   ├── 📄 nodejs_template_config.json
+│   ├── 📁 python_flask_template
+│   │   ├── 📄 python_flask_template_config.json
+│   ├── 📁 java_springboot_template
+│   │   ├── 📄 java_springboot_template_config.json
+├── 📁 examples
+│   ├── 📁 user_management_example
+│   │   ├── 📄 user_management_example_config.yml
+├── 📁 applications
+│   ├── 📁 ecommerce_application
+│   │   ├── 📄 ecommerce_application_config.yml
+```
+
+The item names are purely indicative, this example is only to explain a potential use-case.
+
+The JSON and YAML files above follow the structure explained [in the related documentation](/marketplace/add_to_marketplace/manage_marketplace_items.md).
+
+Here is an example of the Inventory System plugin config file:
+
+```json
+{
+    "name" : "Inventory System",
+    "itemId" : "inventory-system",
+    "description" : "The Inventory System takes charge of a perfect inventory organization.",
+    "type" : "plugin",
+    "categoryId": "",
```

Comment:
Do we need to specify the category or explain how to use it?

Reply:
I am in doubt about it. Given the current restriction I don't know what to explain, maybe we put the category `extracategory`?

Reply:
I agree with extracategory for the moment

Reply:
```suggestion
    "categoryId": "extracategory",
```